recipes/egl/system/conanfile.py

```diff
@@ -0,0 +1,75 @@
+from conans import ConanFile, tools
+from conans.errors import ConanException, ConanInvalidConfiguration
+
+
+class SysConfigEGLConan(ConanFile):
+    name = "egl"
+    version = "system"
+    description = "cross-platform virtual conan package for the EGL support"
+    topics = ("conan", "opengl", "egl")
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://www.khronos.org/egl"
+    license = "MIT"
+    settings = ("os",)
+
+    def configure(self):
+        if self.settings.os == "Macos":
```

Comment:
```suggestion
        if self.settings.os == "Macos":
```

It doesn't restrict Windows, but there is no installation to the same.

Reply:
Yes, because it can be installed and made available to the user on windows  (by using https://github.com/google/angle or amd drivers for example)

I took @SSE4 's [answer above](https://github.com/conan-io/conan-center-index/pull/3665#issuecomment-732108730) `there is no `EGL` for macOS` as `no implementation of the EGL api exist on macOS`, but maybe I misunderstood and he just meant that there was no implementation provided by macOS ?

The question is what do we want for packages linking to EGL, do we want to have 
```
if self.settings.os == "Linux":
    self.requires("egl/system")
```
or 
```
self.requires("egl/system")
```

The former can mislead into thinking that the recipe does not link to egl on non linux platforms, whereas the latter clearly shows that egl is used on all platforms (and drops macos automagically which is the good thing to do if there is no egl implementation there)
But I have no strong opinion on this one, so I'll do as you folks say.

Reply:
yes, there is no system EGL library or framework in OSX. while there are some 3rd-party libraries (such as Angle) providing EGL API, we don't consider them system.

Reply:
ok, So it's basically the same on situation on windows, and I should raise an exception too as @uilianries suggested ?